src/NHibernate/Persister/Entity/AbstractEntityPersister.cs

```diff
@@ -1408,24 +1408,16 @@ public virtual object InitializeLazyProperty(string fieldName, object entity, IS
 			{
 				throw new HibernateException($"Entity is not associated with the session: {id}");
 			}
-
-			int[] indexes;
-			int[] lazyIndexes;
-			if (allLazyProperties)
-			{
-				lazyIndexes = indexes = lazyPropertyNumbers;
```

Comment:
> The problem was with the indexes and lazyIndexes arrays 

What problem do you see with the `indexes`?  I see it only for `lazyIndexes`. Can't fix be  simplified to something like this:
```C#
if (allLazyProperties)
{
	indexes = lazyPropertyNumbers;
	lazyIndexes = new int[lazyPropertyNumbers.Length];
	for (int i = 0; i < lazyIndexes.Length; i++)
	{
		lazyIndexes[i] = i;
	}
}
```

Reply:
You're right; there's no problem with `indexes`. I've simplified the fix.